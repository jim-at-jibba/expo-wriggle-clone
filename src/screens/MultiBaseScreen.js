import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { colors, utilStyles } from '../api/constants';

// grab component
import Touch from '../components/Touch';

const MultiBaseScreen = ({ navigation }) => (
  <View style={styles.container}>
    <ScrollView
      style={styles.container}
      contentContainerStyle={styles.contentContainer}
    >
      <Text style={utilStyles.text}>Multi Screens Content Area</Text>

      <View style={utilStyles.spacer80} />

      <Touch
        onPress={() => navigation.navigate('MultiLevel2')}
        text="go to level 2"
      />
    </ScrollView>
  </View>
);

MultiBaseScreen.navigationOptions = {
  headerTitleStyle: {
    flex: 1,
    textAlign: 'center'
  },
  headerStyle: {
    borderBottomColor: 'transparent',
    elevation: 0
  },
  title: 'Multi Screens'
};

MultiBaseScreen.propTypes = {
  // required
  navigation: PropTypes.object.isRequired
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    flex: 1
  },
  contentContainer: {
    alignItems: 'center',
    paddingTop: 30
  }
});

export default MultiBaseScreen;
