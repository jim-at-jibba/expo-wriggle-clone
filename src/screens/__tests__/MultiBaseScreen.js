import React from 'react';
import renderer from 'react-test-renderer';

import MultiBaseScreen from '../MultiBaseScreen';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn()
};

const tree = renderer
  .create(<MultiBaseScreen navigation={navigation} />)
  .toJSON();

describe('<MultiBaseScreen />', () => {
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
