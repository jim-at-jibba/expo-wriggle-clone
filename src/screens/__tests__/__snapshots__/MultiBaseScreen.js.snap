// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiBaseScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;
